src/Tizen.Network.Stc/Tizen.Network.Stc/StcInfo.cs

```diff
@@ -0,0 +1,281 @@
+                }
+                return incoming;
+            }
+        }
+
+        /// <summary>
+        /// A property to get outgoing counter from statistics information.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        /// <value>outgoing counter.</value>
+        public long OutgoingCounter
+        {
+            get
+            {
+                long incoming;
+                long outgoing;
+                int ret = Interop.Stc.Info.GetCounter(_infoHandle, out incoming, out outgoing);
+                if (ret != (int)StcError.None)
+                {
+                    Log.Error(Globals.LogTag, "Failed to get outgoing counter from info, Error - " + (StcError)ret);
+                }
+                return outgoing;
+            }
+        }
+
+        /// <summary>
+        /// A property to get the roaming type from statistics information.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        /// <value>IsRoaming.</value>
+        public bool IsRoaming
+        {
+            get
+            {
+                RoamingType roaming;
+                int ret = Interop.Stc.Info.GetRoaming(_infoHandle, out roaming);
+                if (ret != (int)StcError.None)
+                {
+                    Log.Error(Globals.LogTag, "Failed to get Roaming type from info, Error - " + (StcError)ret);
+                }
+
+                return roaming == RoamingType.Enabled;
+            }
+        }
+
+        /// <summary>
+        /// To get the network protocol type from statistics information.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        /// <returns>NetworkProtocol type.</returns>
+        /// <exception cref="InvalidOperationException">Thrown when method fails due to an invalid operation/value.</exception>
+        public NetworkProtocol GetProtocol()
+        {
+            NativeNetworkProtocol protocol;
+            int ret = Interop.Stc.Info.GetProtocol(_infoHandle, out protocol);
+            if (ret != (int)StcError.None)
+            {
+                Log.Error(Globals.LogTag, "Failed to get Protocol type from info, Error - " + (StcError)ret);
+            }
+            if (protocol == NativeNetworkProtocol.Unknown)
+            {
+                throw new InvalidOperationException("Network protocol is Unknown.");
+            }
+            return (NetworkProtocol)protocol;
+        }
+
+        /// <summary>
+        /// To get if the app is running as a foreground process, from it's statistics information.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        /// <returns>True, if process state is Foreground, otherwise False.</returns>
+        /// <exception cref="InvalidOperationException">Thrown when method fails due to an invalid operation/value.</exception>
+        public bool IsForegroundProcess()
```

Comment:
Is it possible scenario that throw exception?
Could you explain which case could be occurred error?

Acutally, I want to change it to getter only property without excpetion

Reply:
As suggested in the discussion: Exception throw is removed, Unknown is passed to app as 'Others' and made it a Property.